.github/workflows/test-dev.yml

```
@@ -20,14 +20,98 @@ permissions:
+          echo "BUILD_DEVCONTAINER=true" >> $GITHUB_ENV
+
+          # Make sure the 'devcontainer.json' is up-to-date to predict the next version
+          NEXT_IMAGE_TAG=$(echo "${{ env.VERSION }}" | awk -F. '{print $1"."$2"."$3+1}')
+          sed 's/\/\/.*$//' .devcontainer/devcontainer.json | \
+            grep "\\S" | \
+            yq .build.cacheFrom[0] | \
+            grep -q $NEXT_IMAGE_TAG # Error out if the predicted devcontainer version is not referenced
+
+      - name: The devcontainer does not need to be rebuilt
+        if: steps.changes.outputs.devcontainer != 'true' || steps.extract_branch.outputs.branch == 'main'
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC1009:info:2:105: The mentioned syntax error was in this simple command

(shellcheck)